Source/ACE.Server/WorldObjects/Player_Database.cs

```diff
@@ -18,7 +18,25 @@ partial class Player
         /// This variable is set to true when a change is made, and set to false before a save is requested.<para />
         /// The primary use for this is to trigger save on add/modify/remove of properties.
         /// </summary>
-        public bool CharacterChangesDetected { get; set; }
+        private bool characterChangesDetected;
+        public bool CharacterChangesDetected
```

Comment:
ChangesDetected should still be set even if DoNotSave is set

Reply:
It would always be set by the else case to the value, right? It should only force false if DoNotSave evaluates to true.

Reply:
This logic is not needed, the check is done at the SaveCharacterToDatabase function.